File: javasolutions/TwoSumII.java
import java.util.Arrays;

public class TwoSumII {
  public static int[] findTwoSumII(int[] sortedArry, int sum){
    int low = 0;
    int high = sortedArry.length - 1;

    while(low < high) {
      if (sortedArry[low] + sortedArry[high] > sum) {
        high--;
      } else if (sortedArry[low] + sortedArry[high] < sum) {
        low++;
      } else {
        return new int[]{low + 1, high + 1};
      }
    }

    throw new IllegalArgumentException("No two sum solution.");
  }

  public static void main(String[] arys) {
    int[] test = {1, 2, 3, 5};
    int sum = 5;

    System.out.println(Arrays.toString(TwoSumII.findTwoSumII(test, sum)));
  }
}

File: programming_language/ruby/two_sum_ii.rb
def find_two_sum_ii(sorted_arry, sum)
  low = 0
  high = sorted_arry.length - 1

  while low < high
    if sorted_arry[low] + sorted_arry[high] > sum
      high -= 1
    elsif sorted_arry[low] + sorted_arry[high] < sum
      low += 1
    else
      return [low + 1, high + 1]
    end
  end
end

puts find_two_sum_ii([1, 2, 3, 5], 5)
